adrs/2023-02-10-limit-manager-role-permission.md

```diff
@@ -0,0 +1,93 @@
+- Get/List/Create/Delete/Update/Patch/Watch on `EphemeralRunners` (with `Status` and `Finalizer` sub-resource)
+
+- Get/List/Create/Delete/Update/Patch/Watch on `Pods` (with `Status` sub-resource)
+- **Get/List/Create/Delete/Update/Patch/Watch on `Secrets`**
+- Get/List/Create/Delete/Update/Patch/Watch on `Roles`
+- Get/List/Create/Delete/Update/Patch/Watch on `RoleBindings`
+- Get/List/Create/Delete/Update/Patch/Watch on `ServiceAccounts`
+
+> Full list can be found at: https://github.com/actions/actions-runner-controller/blob/facae69e0b189d3b5dd659f36df8a829516d2896/charts/actions-runner-controller-2/templates/manager_role.yaml
+
+## Limit cluster role permission on Secrets
+
+The cluster scope `List` `Secrets` permission might be a blocker for adopting `actions-runner-controller` for certain customers as they may have certain restriction in their cluster that simply doesn't allow any service account to have cluster scope `List Secrets` permission. 
+
+To help these customers and improve security for `actions-runner-controller` in general, we will try to limit the `ClusterRole` permission of the controller manager's service account down to the following:
+
+- Get/List/Create/Delete/Update/Patch/Watch on `AutoScalingRunnerSets` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `AutoScalingListeners` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `EphemeralRunnerSets` (with `Status` and `Finalizer` sub-resource)
+- Get/List/Create/Delete/Update/Patch/Watch on `EphemeralRunners` (with `Status` and `Finalizer` sub-resource)
+
+- List/Watch on `Pods`
+- List/Watch on `Roles`
+- List/Watch on `RoleBindings`
+- List/Watch on `ServiceAccounts`
+
+> We will change the default cache-based client to bypass cache on reading `Secrets`, so we can eliminate the need for `List` and `Watch` `Secrets` permission in cluster scope.
+
+Introduce a new `ClusterRole` for the controller and `RoleBinding` it to the namespace that each `AutoScalingRunnerSet` deployed with the following permission.
+
+- Get/Create/Delete/Update/Patch on `Secrets`
+- Get/Create/Delete/Update/Patch on `Pods`
+- Get/Create/Delete/Update/Patch on `Roles`
+- Get/Create/Delete/Update/Patch on `RoleBindings`
+- Get/Create/Delete/Update/Patch on `ServiceAccounts`
+
+The `RoleBinding` for the new `ClusterRole` will happen during `helm install demo oci://ghcr.io/actions/actions-runner-controller-charts/auto-scaling-runner-set` to grant the controller's service account required permissions to operate in the namespace the `AutoScalingRunnerSet` deployed.
+
+## Install ARC to only watch/react resources in a single namespace
+
+In case the user doesn't want to have any `ClusterRole`, they can choose to install the `actions-runner-controller` in a mode that only requires a `Role` with `RoleBinding` in a particular namespace.
```

Comment:
@wherka-ama I think this is what you exactly wanted! I'd appreciate your confirmation when you have some time.

Reply:
@mumoshu : that's indeed in line with my expectations! Thank you very much for taking my voice into account. I really appreciate it. 

Looking forward to seeing the implementation.

Cheers

Reply:
@mumoshu , @TingluoHuang : I've just checked a new release - 0.2.0 as per [latest announcements](https://github.com/actions/actions-runner-controller/blob/master/docs/preview/gha-runner-scale-set-controller/README.md):

`helm pull oci://ghcr.io/actions/actions-runner-controller-charts/gha-runner-scale-set-controller     --version 0.2.0 --untar`

I can still see the ClusterRole with ClusterRoleBinding which contains the rules for secrets. There are some improvements of course, like changes around [pv and pvc](https://github.com/actions/actions-runner-controller/pull/2260/files#diff-3b0c76ac2a8b78135a6787687e6d89a69bf9344b1c90c581df6d6d4a45ae10a8)  to name a few, but I'm afraid the secrets are still a bone of contention :/

Nevertheless, thanks a lot for this batch of improvements! We'll wait for the next version to address issues around secrets.

Cheers!

Reply:
@wherka-ama once this ADR is reviewed and merged, I will work on implementing those improvements, which should be included in 0.3.0. 🤞

Reply:
Great work! Thank you very much indeed ❤️

Please do let me know if there is anything in the pipeline for an early preview. I'd be happy to cast an eye and share my thoughts if that's helpful.

Reply:
Once I finish all the changes, you can definitely try it out by building the controller image yourself and using the helm chart from the latest source code. 🙇